### ISSUE TITLE
Rewrite System.Private.Uri tests for desktop

### ISSUE THREAD

**Issue by tarekgh** [dotnet/corefx#15145]
Looks the behavior of netcore is very different than the desktop so the current existing tests are dramatically failing on the desktop and need to be re-written for the desktoop. 

#10821 is an example of failures

**Comment by davidsh** [comment]
This is probably due to quirking issues where the behavior on .NET Framework (Desktop) depends on what TFM is used, i.e. < .NET 4.5 vs >= .NET 4.5.  I know that in the case of #10821 there is a quirk that changes that behavior on Desktop depending on the TFM.

We have lots of tests for Desktop in TFS for Uri.

cc: @CIPop

**Comment by tarekgh** [comment]
I am not sure if the issue is just a quirk. I tried some examples of the test on 4.6 and clearly the behavior is different. and I am seeing the comments on the code like 

```C#
        //This test tests a case where the Core implementation of UriBuilder differs from Desktop Framework UriBuilder.
        //The Query property will not longer prepend a ? character if the string being assigned is already prepended.
```

**Comment by davidsh** [comment]
>        //This test tests a case where the Core implementation of UriBuilder differs from Desktop Framework UriBuilder. //The Query property will not longer prepend a ? character if the string being assigned is already prepended.

For that particular case, yes, the behavior is different.  That is because we changed it on .NET Core but not yet on .NET Framework (Desktop).  It was one of those bugs we proposed as "port-to-netfx" but have not yet acted on the change due to app-compat concerns.

**Comment by CIPop** [comment]
I agree with @davidsh. I don't see this as a rewrite but more as ensuring we're testing against the right quirks and that behavior differences are properly captured by our tests.

**Comment by safern** [comment]
@CIPop @davidsh the problem where is that xunit runner's TFM is 4.5.2 ([see here](https://github.com/xunit/xunit/blob/master/src/xunit.console/App.config)) so all the quirks that are set for later versions will not so we should fix them to set the right switches values when the tests are loaded.

I will disable the tests on Desktop for now.

**Comment by safern** [comment]
The tests that are failing:
System.PrivateUri.Tests.UriBuilderTests.TestQuery
System.PrivateUri.Tests.UriBuilderTests.Fragment_Get_Set_Netcore
System.PrivateUri.Tests.UriRelativeResolutionTest.Uri_Unicode_Format_Character_Combinations_Scheme
System.PrivateUri.Tests.UriRelativeResolutionTest.Uri_Relative_Unicode_Glitchy
System.PrivateUri.Tests.UriRelativeResolutionTest.Uri_Relative_SimplePartialPathWithUnknownScheme_Unicode_ReturnsPartialPathWithScheme
System.PrivateUri.Tests.UriRelativeResolutionTest.Uri_Relative_RightToLeft
System.PrivateUri.Tests.UriRelativeResolutionTest.Uri_Unicode_SurrogatePairs_Scheme
System.PrivateUri.Tests.UriRelativeResolutionTest.Uri_Relative_SimplePartialPathWithScheme_Unicode_ReturnsPartialPathWithScheme

All of this tests are from System.Private.Uri.Funtional.Tests which are being disabled on: https://github.com/dotnet/corefx/pull/18439
